api/app/weather_models/process_grib.py

```diff
@@ -150,8 +151,12 @@ def yield_data_for_stations(self, raster_band: gdal.Dataset):
             x_coordinate, y_coordinate = calculate_raster_coordinate(
                 longitude, latitude, self.padf_transform, self.geo_to_raster_transformer)
 
-            points, values = get_surrounding_grid(
-                raster_band, x_coordinate, y_coordinate)
+            if 0 <= x_coordinate < raster_band.XSize and 0 <= y_coordinate < raster_band.YSize:
```

Comment:
It's a retroactive one. Some of the "active" stations we're getting from the WFWX API are "test stations", with coordinates that are very far from BC. So this check shouldn't really be necessary once we're using the prod version of the WFWX API, but for now it's necessary

Reply:
Ok I see, would it be valuable to add that as a TODO?

Reply:
It's always going to be necessary though - since we don't have direct control over the coordinates of the stations. It's also technically possible that a valid station could be outside of the raster file we're using.